src/components/RouteLayout.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Switch from 'react-router-dom/Switch';
+import Route from 'react-router-dom/Route';
+
+const RouteWithSubRoutes = route => {
+	if (!route.component) {
+		throw new Error(`route for ${route.path} must have a 'component' property`);
+	}
+	if (route.render || route.children) {
+		console.warn('route.render and route.children function not supported');
+	}
+	return (
+		<Route
+			path={route.path}
+			exact={route.exact || false}
+			strict={route.strict || false}
+			render={props => (
+				<route.component {...props}>
+					{route.routes && <SwitchRoutes routes={route.routes} />}
+				</route.component>
+			)}
+		/>
+	);
+};
+
+const SwitchRoutes = ({ routes }) => (
+	<Switch>
+		{routes.map((route, i) =>
+			<RouteWithSubRoutes key={i} {...route} />
```

Comment:
Do we envision a case ever where there's a route that needs a different component or different props than what `RouteWithSubRoutes` provides?